In [1]:
import pandas as pd
import numpy as np

import pickle
import json

#### What's the version of pipenv you installed?
`
pipenv, version 2023.11.15
`

#### What's the first hash for scikit-learn you get in Pipfile.lock?
```
sha256:0c275a06c5190c5ce00af0acbb61c06374087949f643ef32d355ece12c4db043
```

In [4]:
def load_model_and_dv():
    model_file = 'model1.bin'
    dv_file = 'dv.bin'
    model = None
    dv = None
    with open(model_file, 'rb') as f:
        model = pickle.load(f)
    
    with open(dv_file, 'rb') as f:
        dv = pickle.load(f)
    
    return model, dv

In [5]:
model, dv = load_model_and_dv()

D:\Apps\anaconda3\Lib\site-packages\sklearn\base.py:347: InconsistentVersionWarning: Trying to unpickle estimator LogisticRegression from version 1.3.1 when using version 1.3.0. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
D:\Apps\anaconda3\Lib\site-packages\sklearn\base.py:347: InconsistentVersionWarning: Trying to unpickle estimator DictVectorizer from version 1.3.1 when using version 1.3.0. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(


In [6]:
def predict_proba(client):
    X = dv.transform(client)
    return model.predict_proba(X)[:,1][0]

In [7]:
client = {"job": "retired", "duration": 445, "poutcome": "success"}
round(predict_proba(client), 3)

0.902

#### What's the probability that this client will get a credit?

* 0.162
* 0.392
* 0.652
* `0.902`

In [23]:
import requests

url = "http://localhost:8823/predict"
client = {"job": "unknown", "duration": 270, "poutcome": "failure"}
requests.post(url, json=client).json()

0.13968947052356817

#### What's the probability that this client will get a credit?

* `0.140`
* 0.440
* 0.645
* 0.845

Download the base image svizor/zoomcamp-model:3.10.12-slim. You can easily make it by using docker pull command.

#### So what's the size of this base image?

* 47 MB
* `147 MB`
* 374 MB
* 574 MB

### After Docker

In [31]:
url = "http://localhost:8823/predict"
client = {"job": "retired", "duration": 445, "poutcome": "success"}
requests.post(url, json=client).json()

0.726936946355423

#### What's the probability that this client will get a credit now?

* 0.168
* 0.530
* `0.730`
* 0.968